microceph/common/network.go

```diff
@@ -0,0 +1,73 @@
+package common
+
+import (
+	"fmt"
+	"net"
+)
+
+// FindIpOnSubnet scans the host's network interfaces to check if an IP is available
+// for the provided subnet. It returns the found IP address or an empty string.
+func FindIpOnSubnet(subnet string) (string, error) {
+	_, sn, err := net.ParseCIDR(subnet)
+	if err != nil {
+		return "", err
+	}
+
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return "", err
+	}
+
+	for _, iface := range ifaces {
+		addrs, _ := iface.Addrs()
+		for _, addr := range addrs {
+			ipNet, ok := addr.(*net.IPNet)
+			if !ok {
+				continue
+			}
+
+			if !ipNet.IP.IsGlobalUnicast() {
+				continue
+			}
+
+			if sn.Contains(ipNet.IP) {
+				return ipNet.IP.String(), nil
+			}
+		}
+	}
+	return "", fmt.Errorf("no IP belongs to provided subnet %s", subnet)
+}
+
+// FindNetworkAddress locates the provided IP address on host's network interfaces.
+// It returns the containing subnet address on success or an empty string on failure.
+func FindNetworkAddress(address string) (string, error) {
+	// Parse provided address.
+	monIp := net.ParseIP(address)
+	nw := []string{}
+
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return "", err
+	}
+
+	for _, iface := range ifaces {
+		addrs, _ := iface.Addrs()
+		for _, addr := range addrs {
+			ipNet, ok := addr.(*net.IPNet)
+			if !ok {
+				continue
+			}
+
+			if !ipNet.IP.IsGlobalUnicast() {
+				continue
+			}
+
+			nw = append(nw, addr.String())
```

Comment:
The `nw` slice seems unused, is that right?

Reply:
The nw list is used to provided details of available networks that could be provided instead (by the user) in case of incorrect parameters being passed to bootstrap.